File: js/RemoveDuplicatesfromSortedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let current = head;
    while(current && current.next)
        {
            if (current.val == current.val.next) {current.next = current.next.next;}
            else {current = current.next;}
        }
        return head;
};



/*  chatgpt virsion

    var deleteDuplicates = function(head) {
    let current = head;
    
    // Keep going while there's a next node
    while (current && current.next) {
        if (current.val === current.next.val) {
            // Skip the duplicate node
            current.next = current.next.next;
        } else {
            // Move forward
            current = current.next;
        }
    }
    
    return head;
};
*/

File: cpp/RemoveDuplicatesfromSortedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 class Solution {
    public:
    ListNode* deleteDuplicates(ListNode* head) {
        while (head && head.next)
        {
            if (head.val == head.next.val)
            {
                head.next.val == head.next.next.val;
            }
            
            else
            {
                head = head.next;
            }
            
        }
        return head;
    }
};
*/